contracts/metatx/MinimalForwarder.sol

```diff
@@ -26,29 +21,64 @@ contract MinimalForwarder is EIP712 {
         bytes data;
     }
 
-    bytes32 private constant _TYPEHASH =
+    struct ExpiringForwardRequest {
+        uint48 deadline;
+        address from;
+        address to;
+        uint256 value;
+        uint256 gas;
+        uint256 nonce;
+        bytes data;
+    }
+
+    bytes32 private constant _FORWARD_REQUEST_TYPEHASH =
         keccak256("ForwardRequest(address from,address to,uint256 value,uint256 gas,uint256 nonce,bytes data)");
+    bytes32 private constant _EXPIRING_FORWARD_REQUEST_TYPEHASH =
+        keccak256(
+            "ExpiringForwardRequest(uint48 deadline,address from,address to,uint256 value,uint256 gas,uint256 nonce,bytes data)"
+        );
 
     mapping(address => uint256) private _nonces;
 
-    constructor() EIP712("MinimalForwarder", "0.0.1") {}
+    constructor() EIP712(name(), version()) {}
+
+    function version() public view virtual returns (string memory) {
```

Comment:
Why do we have this function ?

Reply:
Customizing EIP712 version, similar to Governor:

https://github.com/OpenZeppelin/openzeppelin-contracts/blob/7b3e7b7055d6e1226d4920ca7d7c446c0c0bfff5/contracts/governance/Governor.sol#L85

I can't see immediate reasons why someone would override it, but I feel it makes sense to leave the option open? What do you think?

Reply:
I can't neither think about a reason why someone would need it, but I'd say the same logic of why we set `virtual` in public functions should apply here.

"Why would someone override `version` and `name`?" is kinda the same question as "Why would someone override X function?" from my perspective.

No strong opinion if we remove it.